### PR TITLE
Documented using external fixed keys for network identity

### PR DESCRIPTION
**Pull request by izolyomi**
As a follow up on issue https://github.com/libp2p/rust-libp2p/issues/2091, I've just added a few sentences on using keys from an external source like an HD wallet or config file.

However, I didn't find related function names too consequent, sometimes it's `decode()`, sometimes `from_bytes()`, even in the same `ed25519` module.

### CODE REVIEW

File: core/src/identity/ed25519.rs [hunk 1]
```diff
@@ -43,8 +43,9 @@ impl Keypair {
         self.0.to_bytes()
     }
 
-    /// Decode a keypair from the format produced by `encode`,
-    /// zeroing the input on success.
+    /// Decode a keypair from the [binary format](https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.5)
+    /// produced by `encode`, zeroing the input on success.
```
Comment (rubdos) [inline core/src/identity/ed25519.rs:47]:
```suggestion
    /// produced by [`Keypair::encode`], zeroing the input on success.
```

while you're at it :-)

File: core/src/identity/ed25519.rs [hunk 2]
```diff
@@ -43,8 +43,9 @@ impl Keypair {
         self.0.to_bytes()
     }
 
-    /// Decode a keypair from the format produced by `encode`,
-    /// zeroing the input on success.
+    /// Decode a keypair from the [binary format](https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.5)
+    /// produced by [`Keypair::encode`], zeroing the input on success.
```
Comment (mxinden) [inline core/src/identity/ed25519.rs:47]:
```suggestion
    /// produced by [`Keypair::encode`], zeroing the input on success.
    ///
```

File: core/src/identity/secp256k1.rs [hunk 3]
```diff
@@ -83,14 +83,15 @@ impl fmt::Debug for SecretKey {
...
-    /// Generate a new Secp256k1 secret key.
+    /// Generate a new random Secp256k1 secret key.
     pub fn generate() -> SecretKey {
         SecretKey(libsecp256k1::SecretKey::random(&mut rand::thread_rng()))
     }
 
     /// Create a secret key from a byte slice, zeroing the slice on success.
     /// If the bytes do not constitute a valid Secp256k1 secret key, an
     /// error is returned.
+    /// Note that the expected binary format is the same as `libsecp256k1`'s.
```
Comment (mxinden) [inline core/src/identity/secp256k1.rs:94]:
```suggestion
    ///
    /// Note that the expected binary format is the same as `libsecp256k1`'s.
```

**Review by mxinden — APPROVED** [review]
Spot on. Thank you very much @izolyomi. As someone working with the codebase day-to-day I don't have an eye for what problems people new to the codebase face. Pull requests like this one are very helpful.

**Pull request merged**